### DISCUSSION TITLE
Where is the effective dependency on `cargo-xtask` documented?

### DISCUSSION THREAD

**Discussion by sanmai-NL** [aya-rs/aya#830] [Q&A] [answered]
Where is the effective dependency on `cargo-xtask` documented?

**Comment by tamird** [comment, marked as answer]
Probably nowhere. Note that it's a development-only dependency.

**Reply by sanmai-NL** [reply]
Do or can the instructions at https://github.com/aya-rs/bpf-linker#nightly replace the `xtask` dependency? I'd rather remove it.

**Reply by tamird** [reply]
Your question suggests you haven't spent time looking at the functionality implemented using xtask.

I think we'd have a more productive conversation over a PR.

**Reply by sanmai-NL** [reply]
I had. But in my-cargo-generated product the only thing it seems to do extra is remove the `RUSTUP_TOOLCHAIN` env var. Can you give an example of something I missed and where I had to look to find that out? Maybe I discover the additional functionalities as well after your pointer.

I'm not sure I understand your PR suggestion. What should be changed then? For me, I first want to establish the necessity of `xtask` before I go on and create a PR to remove it.

**Reply by alessandrod** [reply]
If you generated a project with cargo-generate and aya-template, you can absolutely remove xtask. It's just a convenience wrapper over having to run cargo build manually. If you don't use it it won't do anything. Feel free to remove it if you don't have use for it.

**Reply by sanmai-NL** [reply]
Ahh, I think we misunderstood each other. I'm fine with the `xtask` alias, just questioned the `xtask` crate from crates.io. It turns out, the latter isn't a dependency at all. I read `cargo xtask` in the build instructions and of course, a standard invocation of `cargo-xtask` looks like that. And indeed, I found the crate named as such. But the mechanism here is through the alias.

This illustrates why naming different things the same will cause confusion.
